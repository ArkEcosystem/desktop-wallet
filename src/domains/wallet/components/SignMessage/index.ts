export * from "./LedgerReview";
export * from "./SignMessage";
