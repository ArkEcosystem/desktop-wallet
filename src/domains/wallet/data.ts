export const wallets = [
	{
		coinIcon: "Ark",
		coinClassName: "text-theme-danger-400 border-theme-danger-200",
		avatarId: "test1",
		address: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
		publicKey: "028fe98e42e159f2450a52371dfb23ae69a39fc5fee6545690b7f51bfcee933357",
		walletName: "ARK Wallet 1",
		balance: "120 ARK",
		fiat: "980 USD",
		walletTypeIcons: ["Multisig", "Ledger", "Key", "Bridgechain"],
		actions: [
			{
				label: "Action 1",
				value: "1",
			},
			{
				label: "Action 2",
				value: "2",
			},
			{
				label: "Action 3",
				value: "3",
			},
		],
		delegates: [
			{
				username: "Delegate 3",
				address: "AbuzhuDTyhnfAqepZzVcVsgd1Ym6FgETuW",
				rank: 1,
				explorerUrl: "https://dexplorer.ark.io",
				msqUrl: "https://marketsquare.ark.io",
				isActive: true,
			},
		],
		business: {
			name: "ROBank Eco",
		},
		pendingTransactions: [
			{
				id: "ee4175091d9f4dacf5fed213711c3e0e4cc371e37afa7bce0429d09bcf3ecefe",
				confirmations: "100",
				timestamp: "17 Mar 2020 10:22:05",
				type: "transfer",
				sender: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
				recipient: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
				amount: "52",
				fee: "0.2",
				vendorField: "Test",
				isSent: true,
				isMultiSignature: true,
				isSignaturePending: true,
			},
		],
		transactions: [
			{
				id: "ee4175091d9f4dacf5fed213711c3e0e4cc371e37afa7bce0429d09bcf3ecefe",
				confirmations: "100",
				timestamp: "17 Mar 2020 10:22:05",
				type: "delegateRegistration",
				sender: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
				recipient: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
				amount: "0",
				fee: "0.1",
				isSent: true,
			},
			{
				id: "ee4175091d9f4dacf5fed213711c3e0e4cc371e37afa7bce0429d09bcf3ecefe",
				confirmations: "100",
				timestamp: "17 Mar 2020 10:22:05",
				type: "secondSignature",
				sender: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
				recipient: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
				amount: "0",
				fee: "0.1",
				isSent: true,
			},
			{
				id: "ee4175091d9f4dacf5fed213711c3e0e4cc371e37afa7bce0429d09bcf3ecefe",
				confirmations: "100",
				timestamp: "17 Mar 2020 10:22:05",
				type: "businessRegistration",
				sender: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
				recipient: "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT",
				amount: "0",
				fee: "0.1",
				isSent: true,
			},
		],
		hasStarred: true,
	},
];

export const networks = [
	{
		icon: "Ark",
		name: "ARK Ecosystem",
		className: "text-theme-danger-400 border-theme-danger-light",
	},
	{
		icon: "Bitcoin",
		name: "Bitcoin",
		className: "text-theme-warning-400 border-theme-warning-200",
	},
	{
		icon: "Ethereum",
		name: "Ethereum",
		className: "text-theme-neutral-800 border-theme-neutral-600",
	},
	{
		icon: "Lisk",
		name: "Lisk",
		className: "text-theme-primary border-theme-primary-400",
	},
	{
		icon: "Ripple",
		name: "Ripple",
		className: "text-theme-primary-dark border-theme-primary-500",
	},
];
