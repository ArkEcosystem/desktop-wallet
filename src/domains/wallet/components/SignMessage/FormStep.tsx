import { ReadWriteWallet } from "@arkecosystem/platform-sdk-profiles";
import { Avatar } from "app/components/Avatar";
import { FormField, FormLabel } from "app/components/Form";
import { Header } from "app/components/Header";
import { Input, InputAddonStart, InputDefault, InputGroup, InputPassword } from "app/components/Input";
import { useValidation } from "app/hooks";
import React, { ChangeEvent } from "react";
import { useFormContext } from "react-hook-form";
import { useTranslation } from "react-i18next";

export const FormStep = ({ wallet }: { wallet: ReadWriteWallet }) => {
	const { t } = useTranslation();

	const { authentication } = useValidation();

	const { register, setValue } = useFormContext();

	return (
		<section className="space-y-8">
			<Header
				title={t("WALLETS.MODAL_SIGN_MESSAGE.FORM_STEP.TITLE")}
				subtitle={t("WALLETS.MODAL_SIGN_MESSAGE.FORM_STEP.DESCRIPTION")}
			/>

			<FormField name="signatory-address">
				<FormLabel label={t("WALLETS.SIGNATORY")} />
				<InputGroup>
					<InputAddonStart>
						<Avatar address={wallet.address()} size="sm" className="ml-4" noShadow />
					</InputAddonStart>
					<Input value={wallet.address()} className="font-semibold pl-15" disabled />
				</InputGroup>
			</FormField>

			<FormField name="message">
				<FormLabel label={t("COMMON.MESSAGE")} />
				<InputDefault
					ref={register({
						required: t("COMMON.VALIDATION.FIELD_REQUIRED", {
							field: t("COMMON.MESSAGE"),
						}).toString(),
					})}
					onChange={(event: ChangeEvent<HTMLInputElement>) =>
						setValue("message", event.target.value, {
							shouldDirty: true,
							shouldValidate: true,
						})
					}
					data-testid="SignMessage__message-input"
				/>
			</FormField>

			{!wallet.isLedger() && (
				<FormField name="mnemonic">
					<FormLabel label={t("COMMON.YOUR_PASSPHRASE")} />
					<InputPassword
						ref={register(authentication.mnemonic(wallet.coin(), wallet.address()))}
						data-testid="SignMessage__mnemonic-input"
					/>
				</FormField>
			)}
		</section>
	);
};
