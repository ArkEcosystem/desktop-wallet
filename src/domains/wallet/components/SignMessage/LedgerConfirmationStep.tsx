import { Header } from "app/components/Header";
import { LedgerConfirmation } from "domains/transaction/components/LedgerConfirmation";
import { TransactionDetail } from "domains/transaction/components/TransactionDetail";
import React from "react";
import { useTranslation } from "react-i18next";

export const LedgerConfirmationStep = ({ message }: { message: string }) => {
	const { t } = useTranslation();

	return (
		<>
			<Header title={t("WALLETS.MODAL_SIGN_MESSAGE.LEDGER_CONFIRMATION_STEP.TITLE")} />

			<LedgerConfirmation detailsHeading={null}>
				<TransactionDetail label={t("COMMON.MESSAGE")} padding={false} border={false}>
					<span className="break-all">{message}</span>
				</TransactionDetail>
			</LedgerConfirmation>
		</>
	);
};
