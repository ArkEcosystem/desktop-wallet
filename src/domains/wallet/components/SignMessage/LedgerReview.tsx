import { LedgerConfirmation } from "domains/transaction/components/LedgerConfirmation";
import { TransactionDetail } from "domains/transaction/components/TransactionDetail";
import React, { useState } from "react";
import { useFormContext } from "react-hook-form";
import { useTranslation } from "react-i18next";

export const LedgerReview = () => {
	const { getValues, watch } = useFormContext();

	// getValues does not get the value of `defaultValues` on first render
	const [watched] = useState(() => watch());
	const message = getValues("message") || watched.message;

	const { t } = useTranslation();

	return (
		<LedgerConfirmation detailsHeading={null}>
			<TransactionDetail label={t("COMMON.MESSAGE")} padding={false} border={false}>
				<span className="break-all">{message}</span>
			</TransactionDetail>
		</LedgerConfirmation>
	);
};
