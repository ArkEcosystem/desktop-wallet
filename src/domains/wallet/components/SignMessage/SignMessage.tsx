// UI Elements
import { Address } from "app/components/Address";
import { Avatar } from "app/components/Avatar";
import { Button } from "app/components/Button";
import { Circle } from "app/components/Circle";
import { Form, FormField, FormHelperText, FormLabel } from "app/components/Form";
import { Icon } from "app/components/Icon";
import { Input, InputPassword } from "app/components/Input";
import { Modal } from "app/components/Modal";
import { TextArea } from "app/components/TextArea";
import { TransactionDetail } from "app/components/TransactionDetail";
import React, { createRef } from "react";
import { useForm } from "react-hook-form";

type Props = {
	onSubmit?: any;
	isOpen: boolean;
	isSigned?: boolean;
	handleClose?: any;
	handleSign?: any;
	signatoryAddress: string;
};

const mockSignature = {
	publicKey: "03basda32b03s2df1as0csd32f1asd0asd21sv0asd5sa3d1as03d2as",
	signature:
		"304444354032424a3sd210a3s2d4as3d54as0321das03d777cas32f1as3d21as03v5ds4sd3asc045as35d4a03x24z3c5z4sd3as054cas65xv4cxz3v24zs30x24c3xz2v4zs03d54czdv54z3x5c40zx6v54z3df2c4zd0",
	message: "Xp879878687z6xc876Z*6cz87c68zx76c8x7zc68zx7cvsa7dc5as8d765as87d5sa8d7as65dsadasdsad",
};

export const SignMessage = ({ onSubmit, signatoryAddress, isOpen, isSigned, handleClose, handleSign }: Props) => {
	const form = useForm();
	const { register } = form;
	const messageRef = createRef();

	const SignForm = (
		<Form id="sign-message__form" context={form} onSubmit={onSubmit}>
			<FormField name="signatory-address">
				<FormLabel label="Signatory" />
				<div className="relative">
					<Input type="text" disabled />
					<div className="absolute top-0 flex items-center mt-2 ml-4">
						<div className="flex items-center">
							<Avatar address="test" size="sm" noShadow />
							<span className="ml-3 font-semibold ">{signatoryAddress}</span>
						</div>
					</div>
				</div>
			</FormField>
			<FormField name="message">
				<FormLabel label="Message" />
				<Input type="text" ref={register({ required: true })} />
				<FormHelperText />
			</FormField>
			<FormField name="passphrase">
				<FormLabel label="Your Passphrase" />
				<InputPassword ref={register({ required: true })} />
				<FormHelperText />
			</FormField>
			<div className="flex justify-end space-x-3">
				<Button variant="plain">Cancel</Button>
				<Button data-testid="sign-message__sign-button" onClick={() => handleSign()}>
					Sign
				</Button>
			</div>
		</Form>
	);

	const MessageSigned = (
		<div>
			<TransactionDetail
				border={false}
				label="Signatory"
				extra={
					<div className="flex items-center">
						<Circle className="-mr-2 border-black">
							<Icon name="Delegate" width={25} height={25} />
						</Circle>
						<Avatar address="test" size="sm" />
					</div>
				}
			>
				<Address address="AUexKjGtgsSpVzPLs6jNMM6vJ6znEVTQWK" walletName={"ROBank"} />
			</TransactionDetail>
			<TransactionDetail border label="Message" className="text-lg">
				Oleg Happy in the Oleg Bank
			</TransactionDetail>
			<TransactionDetail border label="Signature">
				<TextArea
					className="mt-2 rounded-lg"
					name="signature"
					wrap="hard"
					ref={messageRef}
					defaultValue={JSON.stringify(mockSignature)}
				/>
			</TransactionDetail>

			<div className="flex justify-end pb-5 mt-3">
				<Button variant="plain">
					<Icon name="Copy" />
					<span>Copy Signature</span>
				</Button>
			</div>
		</div>
	);

	const renderSignedMessageContent = () => (isSigned ? MessageSigned : SignForm);

	return (
		<Modal
			isOpen={isOpen}
			title={!isSigned ? "Sign Message" : "Message Successfully Signed"}
			description={!isSigned ? "Insert a message below to sign using your private key" : ""}
			onClose={() => handleClose()}
		>
			<div className="mt-2">{renderSignedMessageContent()}</div>
		</Modal>
	);
};

SignMessage.defaultProps = {
	isOpen: false,
	isSigned: false,
};
