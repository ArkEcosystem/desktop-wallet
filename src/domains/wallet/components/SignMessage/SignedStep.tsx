import { SignedMessage } from "@arkecosystem/platform-sdk/dist/contracts";
import { Address } from "app/components/Address";
import { Avatar } from "app/components/Avatar";
import { Header } from "app/components/Header";
import { TextArea } from "app/components/TextArea";
import { TransactionDetail } from "domains/transaction/components/TransactionDetail";
import React, { useRef } from "react";
import { useTranslation } from "react-i18next";

export const SignedStep = ({ signedMessage }: { signedMessage: SignedMessage }) => {
	const { t } = useTranslation();

	const messageRef = useRef();

	return (
		<section>
			<Header title={t("WALLETS.MODAL_SIGN_MESSAGE.SIGNED_STEP.TITLE")} />

			<TransactionDetail
				border={false}
				label={t("WALLETS.SIGNATORY")}
				extra={<Avatar size="lg" address={signedMessage.signatory} />}
			>
				<Address address={signedMessage.signatory} />
			</TransactionDetail>

			<TransactionDetail label={t("COMMON.MESSAGE")} className="text-lg break-all">
				{signedMessage.message}
			</TransactionDetail>

			<TransactionDetail label={t("COMMON.SIGNATURE")} paddingPosition="top">
				<TextArea
					className="mt-2 rounded-lg"
					name="signature"
					wrap="hard"
					ref={messageRef}
					defaultValue={JSON.stringify(signedMessage)}
				/>
			</TransactionDetail>
		</section>
	);
};
