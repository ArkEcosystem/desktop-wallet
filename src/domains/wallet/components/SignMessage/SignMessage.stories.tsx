import { ARK } from "@arkecosystem/platform-sdk-ark";
import { Environment } from "@arkecosystem/platform-sdk-profiles";
import { boolean, text, withKnobs } from "@storybook/addon-knobs";
import { EnvironmentProvider } from "app/contexts";
import { httpClient } from "app/services";
import React from "react";
import { StubStorage } from "tests/mocks";

import { SignMessage } from "./SignMessage";

const env = new Environment({ coins: { ARK }, httpClient, storage: new StubStorage() });
const profile = env.profiles().create("Anne Doe");

export default { title: "Domains / Wallet / Components / Sign Message", decorators: [withKnobs] };

export const Default = () => {
	const signatoryAddress = text("Address", "ASuusXSW9kfWnicScSgUTjttP6T9GQ3kqT");
	const isOpen = boolean("Is Open?", true);

	return (
		<EnvironmentProvider env={env}>
			<SignMessage
				profileId={profile.id()}
				walletId={"123"}
				signatoryAddress={signatoryAddress}
				isOpen={isOpen}
				onClose={() => alert("Close action")}
				onCancel={() => alert("Cancel action")}
			/>
		</EnvironmentProvider>
	);
};
