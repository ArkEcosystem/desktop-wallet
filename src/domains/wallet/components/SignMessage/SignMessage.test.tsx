/* eslint-disable @typescript-eslint/require-await */
import { ARK } from "@arkecosystem/platform-sdk-ark";
import { Environment, Profile, Wallet } from "@arkecosystem/platform-sdk-profiles";
import { EnvironmentProvider } from "app/contexts";
import { httpClient } from "app/services";
import nock from "nock";
import React from "react";
import { act, fireEvent, render, RenderResult, waitFor } from "testing-library";
import { profiles } from "tests/fixtures/env/data";
import { identity } from "tests/fixtures/identity";
import { StubStorage } from "tests/mocks";

import { translations } from "../../i18n";
import { SignMessage } from "./SignMessage";

let env: Environment;
let profile: Profile;
let wallet: Wallet;

describe("SignMessage", () => {
	beforeAll(() => {
		nock.disableNetConnect();

		nock("https://dwallets.ark.io")
			.get("/api/node/configuration")
			.reply(200, require("../../../../tests/fixtures/coins/ark/configuration-devnet.json"))
			.get("/api/peers")
			.reply(200, require("../../../../tests/fixtures/coins/ark/peers.json"))
			.get("/api/node/configuration/crypto")
			.reply(200, require("../../../../tests/fixtures/coins/ark/cryptoConfiguration.json"))
			.get("/api/node/syncing")
			.reply(200, require("../../../../tests/fixtures/coins/ark/syncing.json"))
			.get("/api/wallets/D61mfSggzbvQgTUe6JhYKH2doHaqJ3Dyib")
			.reply(200, require("../../../../tests/fixtures/coins/ark/wallet.json"))
			.persist();
	});

	beforeEach(async () => {
		env = new Environment({ coins: { ARK }, httpClient, storage: new StubStorage() });

		await env.bootFromObject({ data: {}, profiles });

		profile = env.profiles().findById("bob");

		wallet = await profile.wallets().importByMnemonic(identity.mnemonic, "ARK", "devnet");
	});

	it("should render", () => {
		const { asFragment } = render(
			<EnvironmentProvider env={env}>
				<SignMessage
					profileId={profile.id()}
					walletId={wallet.id()}
					signatoryAddress="AUexKjGtgsSpVzPLs6jNMM6vJ6znEVTQWK"
					isOpen={true}
				/>
			</EnvironmentProvider>,
		);

		expect(asFragment()).toMatchSnapshot();
	});

	it("should sign message", async () => {
		const onSubmit = jest.fn();
		let rendered: RenderResult;

		await act(async () => {
			rendered = render(
				<EnvironmentProvider env={env}>
					<SignMessage
						profileId={profile.id()}
						walletId={wallet.id()}
						signatoryAddress="AUexKjGtgsSpVzPLs6jNMM6vJ6znEVTQWK"
						isOpen={true}
						onSubmit={onSubmit}
					/>
				</EnvironmentProvider>,
			);

			await waitFor(() =>
				expect(rendered.getByTestId("modal__inner")).toHaveTextContent(translations.MODAL_SIGN_MESSAGE.TITLE),
			);
		});

		const { getByTestId, asFragment } = rendered;

		expect(asFragment()).toMatchSnapshot();

		await act(async () => {
			const messageText = "Hello World";
			const signature =
				"304402200fb4adddd1f1d652b544ea6ab62828a0a65b712ed447e2538db0caebfa68929e02205ecb2e1c63b29879c2ecf1255db506d671c8b3fa6017f67cfd1bf07e6edd1cc8";

			const messageInput = getByTestId("SignMessage__message-input");
			expect(messageInput).toBeTruthy();

			await fireEvent.change(messageInput, { target: { value: messageText } });

			const mnemonicInput = getByTestId("SignMessage__mnemonic-input");
			expect(mnemonicInput).toBeTruthy();

			await fireEvent.change(mnemonicInput, { target: { value: identity.mnemonic } });

			await fireEvent.click(getByTestId("SignMessage__submit-button"));

			await waitFor(() => {
				expect(onSubmit).toBeCalledWith({ message: messageText, signatory: wallet.publicKey(), signature });
			});

			expect(rendered.getByTestId("modal__inner")).toHaveTextContent(
				translations.MODAL_SIGN_MESSAGE.SUCCESS_TITLE,
			);
		});
	});
});
