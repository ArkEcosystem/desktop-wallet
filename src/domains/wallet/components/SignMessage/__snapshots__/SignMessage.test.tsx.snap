// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignMessage should render 1`] = `
<DocumentFragment>
  <div
    class="fixed inset-0 z-50 bg-black opacity-50"
    data-testid="modal__overlay"
  />
  <div
    class="sc-fznyAO ioJhAT fixed top-0 left-0 right-0 z-50 flex flex-col px-10 pt-6 pb-8 mx-auto mt-24 overflow-hidden rounded-xl bg-theme-background"
    data-testid="modal__inner"
  >
    <div
      class="absolute top-0 right-0 z-50 mt-5 mr-5"
    >
      <button
        class="sc-AxirZ iJKGpA"
        color="neutral"
        data-testid="modal__close-btn"
        type="button"
      >
        <div
          class="p-1"
        >
          <div
            class="sc-AxhUy iFvBPK"
            height="12"
            width="12"
          >
            <svg>
              cross-slim.svg
            </svg>
          </div>
        </div>
      </button>
    </div>
    <div
      class="py-4"
    >
      <h2
        class="mb-0 text-3xl font-bold"
      >
        Sign Message
      </h2>
      <div
        class="flex-1"
      >
        <div
          class="mt-1 text-theme-neutral-dark"
        >
          Insert a message below to sign using your private key
        </div>
        <div
          class="mt-8"
        >
          <form
            class="space-y-8"
            data-testid="SignMessage__form"
          >
            <fieldset
              class="sc-AxhCb ckQmOL flex flex-col space-y-2"
            >
              <label
                class="flex inline-block mb-2 text-sm font-medium FormLabel transition-colors duration-100 text-theme-neutral-dark"
                data-testid="FormLabel"
                for="signatory-address"
                label="Signatory"
              >
                Signatory
              </label>
              <div
                class="relative"
              >
                <input
                  class="sc-AxgMl jRvGoE overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-neutral-300 text-theme-neutral-900 transition-colors duration-200 px-4 py-3 "
                  data-testid="Input"
                  disabled=""
                  name="signatory-address"
                  type="text"
                />
                <div
                  class="absolute top-0 flex items-center mt-2 ml-4"
                >
                  <div
                    class="flex items-center"
                  >
                    <div
                      class="Avatar__AvatarWrapper-sc-1vsy2s2-0 laNXvc"
                      data-testid="Avatar"
                    >
                      <div
                        class="w-full h-full"
                      >
                        <img
                          alt="test"
                          src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(255, 87, 34)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"45\\" cx=\\"60\\" cy=\\"30\\" fill=\\"rgb(3, 169, 244)\\"/><circle r=\\"35\\" cx=\\"70\\" cy=\\"40\\" fill=\\"rgb(156, 39, 176)\\"/><circle r=\\"55\\" cx=\\"90\\" cy=\\"70\\" fill=\\"rgb(33, 150, 243)\\"/></svg>"
                          title="test"
                        />
                      </div>
                    </div>
                    <span
                      class="ml-3 font-semibold "
                    >
                      D61mfSggzbvQgTUe6JhYKH2doHaqJ3Dyib
                    </span>
                  </div>
                </div>
              </div>
            </fieldset>
            <fieldset
              class="sc-AxhCb ckQmOL flex flex-col space-y-2"
            >
              <label
                class="flex inline-block mb-2 text-sm font-medium FormLabel transition-colors duration-100 text-theme-neutral-dark"
                data-testid="FormLabel"
                for="message"
                label="Message"
              >
                Message
              </label>
              <input
                class="sc-AxgMl jRvGoE overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-neutral-300 text-theme-neutral-900 transition-colors duration-200 px-4 py-3 "
                data-testid="SignMessage__message-input"
                name="message"
                type="text"
              />
            </fieldset>
            <fieldset
              class="sc-AxhCb ckQmOL flex flex-col space-y-2"
            >
              <label
                class="flex inline-block mb-2 text-sm font-medium FormLabel transition-colors duration-100 text-theme-neutral-dark"
                data-testid="FormLabel"
                for="mnemonic"
                label="Your Passphrase"
              >
                Your Passphrase
              </label>
              <div
                class="sc-fzozJi OgXfh max-w-20"
              >
                <input
                  class="sc-AxgMl jRvGoE overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-neutral-300 text-theme-neutral-900 transition-colors duration-200 px-4 py-3 pr-12"
                  data-testid="SignMessage__mnemonic-input"
                  name="mnemonic"
                  type="password"
                />
                <div
                  class="sc-AxheI sc-AxmLO jcthaj my-px mr-4"
                >
                  <button
                    class="flex items-center justify-center w-full h-full text-2xl focus:outline-none bg-theme-background text-theme-primary-400"
                    data-testid="InputPassword__toggle"
                    type="button"
                  >
                    <div
                      class="sc-AxhUy cPXLjK"
                      height="1em"
                      width="1em"
                    >
                      <svg>
                        eye.svg
                      </svg>
                    </div>
                  </button>
                </div>
              </div>
            </fieldset>
            <div
              class="flex justify-end space-x-3"
            >
              <button
                class="sc-AxirZ fhzkRl"
                color="primary"
                data-testid="SignMessage__cancel"
                type="button"
              >
                Cancel
              </button>
              <button
                class="sc-AxirZ dPtiuf"
                color="primary"
                data-testid="SignMessage__submit-button"
                type="submit"
              >
                Sign
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SignMessage should sign message 1`] = `
<DocumentFragment>
  <div
    class="fixed inset-0 z-50 bg-black opacity-50"
    data-testid="modal__overlay"
  />
  <div
    class="sc-fznyAO ioJhAT fixed top-0 left-0 right-0 z-50 flex flex-col px-10 pt-6 pb-8 mx-auto mt-24 overflow-hidden rounded-xl bg-theme-background"
    data-testid="modal__inner"
  >
    <div
      class="absolute top-0 right-0 z-50 mt-5 mr-5"
    >
      <button
        class="sc-AxirZ iJKGpA"
        color="neutral"
        data-testid="modal__close-btn"
        type="button"
      >
        <div
          class="p-1"
        >
          <div
            class="sc-AxhUy iFvBPK"
            height="12"
            width="12"
          >
            <svg>
              cross-slim.svg
            </svg>
          </div>
        </div>
      </button>
    </div>
    <div
      class="py-4"
    >
      <h2
        class="mb-0 text-3xl font-bold"
      >
        Sign Message
      </h2>
      <div
        class="flex-1"
      >
        <div
          class="mt-1 text-theme-neutral-dark"
        >
          Insert a message below to sign using your private key
        </div>
        <div
          class="mt-8"
        >
          <form
            class="space-y-8"
            data-testid="SignMessage__form"
          >
            <fieldset
              class="sc-AxhCb ckQmOL flex flex-col space-y-2"
            >
              <label
                class="flex inline-block mb-2 text-sm font-medium FormLabel transition-colors duration-100 text-theme-neutral-dark"
                data-testid="FormLabel"
                for="signatory-address"
                label="Signatory"
              >
                Signatory
              </label>
              <div
                class="relative"
              >
                <input
                  class="sc-AxgMl jRvGoE overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-neutral-300 text-theme-neutral-900 transition-colors duration-200 px-4 py-3 "
                  data-testid="Input"
                  disabled=""
                  name="signatory-address"
                  type="text"
                />
                <div
                  class="absolute top-0 flex items-center mt-2 ml-4"
                >
                  <div
                    class="flex items-center"
                  >
                    <div
                      class="Avatar__AvatarWrapper-sc-1vsy2s2-0 laNXvc"
                      data-testid="Avatar"
                    >
                      <div
                        class="w-full h-full"
                      >
                        <img
                          alt="test"
                          src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(255, 87, 34)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"45\\" cx=\\"60\\" cy=\\"30\\" fill=\\"rgb(3, 169, 244)\\"/><circle r=\\"35\\" cx=\\"70\\" cy=\\"40\\" fill=\\"rgb(156, 39, 176)\\"/><circle r=\\"55\\" cx=\\"90\\" cy=\\"70\\" fill=\\"rgb(33, 150, 243)\\"/></svg>"
                          title="test"
                        />
                      </div>
                    </div>
                    <span
                      class="ml-3 font-semibold "
                    >
                      D61mfSggzbvQgTUe6JhYKH2doHaqJ3Dyib
                    </span>
                  </div>
                </div>
              </div>
            </fieldset>
            <fieldset
              class="sc-AxhCb ckQmOL flex flex-col space-y-2"
            >
              <label
                class="flex inline-block mb-2 text-sm font-medium FormLabel transition-colors duration-100 text-theme-neutral-dark"
                data-testid="FormLabel"
                for="message"
                label="Message"
              >
                Message
              </label>
              <input
                class="sc-AxgMl jRvGoE overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-neutral-300 text-theme-neutral-900 transition-colors duration-200 px-4 py-3 "
                data-testid="SignMessage__message-input"
                name="message"
                type="text"
              />
            </fieldset>
            <fieldset
              class="sc-AxhCb ckQmOL flex flex-col space-y-2"
            >
              <label
                class="flex inline-block mb-2 text-sm font-medium FormLabel transition-colors duration-100 text-theme-neutral-dark"
                data-testid="FormLabel"
                for="mnemonic"
                label="Your Passphrase"
              >
                Your Passphrase
              </label>
              <div
                class="sc-fzozJi OgXfh max-w-20"
              >
                <input
                  class="sc-AxgMl jRvGoE overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-neutral-300 text-theme-neutral-900 transition-colors duration-200 px-4 py-3 pr-12"
                  data-testid="SignMessage__mnemonic-input"
                  name="mnemonic"
                  type="password"
                />
                <div
                  class="sc-AxheI sc-AxmLO jcthaj my-px mr-4"
                >
                  <button
                    class="flex items-center justify-center w-full h-full text-2xl focus:outline-none bg-theme-background text-theme-primary-400"
                    data-testid="InputPassword__toggle"
                    type="button"
                  >
                    <div
                      class="sc-AxhUy cPXLjK"
                      height="1em"
                      width="1em"
                    >
                      <svg>
                        eye.svg
                      </svg>
                    </div>
                  </button>
                </div>
              </div>
            </fieldset>
            <div
              class="flex justify-end space-x-3"
            >
              <button
                class="sc-AxirZ fhzkRl"
                color="primary"
                data-testid="SignMessage__cancel"
                type="button"
              >
                Cancel
              </button>
              <button
                class="sc-AxirZ dPtiuf"
                color="primary"
                data-testid="SignMessage__submit-button"
                type="submit"
              >
                Sign
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
